src/EditorFeatures/TestUtilities/Threading/WpfTestCase.cs

```diff
@@ -83,5 +87,20 @@ public override Task<RunSummary> RunAsync(IMessageSink diagnosticMessageSink, IM
 
             return task.Unwrap();
         }
+
+        private static string s_wpfFactRequirementReason;
+
+        /// <summary>
+        /// Asserts that the test is running on a <see cref="WpfFactAttribute"/> test method, and records the reason for requiring the <see cref="WpfFactAttribute"/>.
+        /// </summary>
+        public static void RequireWpfFact(string reason)
+        {
+            if (ForegroundThreadDataInfo.CurrentForegroundThreadDataKind != ForegroundThreadDataKind.StaUnitTest)
+            {
+                throw new Exception($"This test requires {nameof(WpfFactAttribute)} because '{reason}' but is missing {nameof(WpfFactAttribute)}. Either the attribute should be changed, or the reason it needs an STA thread audited.");
+            }
+
+            s_wpfFactRequirementReason = reason;
```

Comment:
It's not in this PR, but I have an upcoming change that does, and fails the test if it's not set. That change is actually going in *last* because right now people are adding new tests with WpfFact and until they stop that's almost assured to break the build.